bndtools.m2e/src/bndtools/m2e/BndConfigurator.java

```diff
@@ -140,10 +140,13 @@ public AbstractBuildParticipant getBuildParticipant(final IMavenProjectFacade pr
 
 				final IProject project = projectFacade.getProject();
 
-				// check if we need to run. After the jars get build, opened or
-				// just viewed we might be called and another run can cause a
-				// build loop
-				if (!needsBuilding(getDelta(project), projectFacade)) {
+				// Check if we need to run. If extensions are enabled then we
+				// don't need to run since bnd-maven-plugin is doing the
+				// packaging. After the jars get build, opened or just viewed we
+				// might be called and another run can cause a build loop
+				if (execution.getPlugin()
+					.isExtensions() || !needsBuilding(getDelta(project), projectFacade)) {
```

Comment:
Short circuiting on `isExtensions()` maybe too broad. One could configure extension in a parent pom pluginManagement, but a child project could use the old way.

Reply:
I think the solution here is to discover if there is a `bnd-maven-plugin:(test-)jar` goal that matches by classifier in the same project and only short circuit in that case.